yarn-project/merkle-tree/src/merkle_tree.ts

```diff
@@ -35,6 +36,16 @@ export interface MerkleTree extends SiblingPathSource {
    * Commit pending updates to the tree
    */
   commit(): Promise<void>;
+  /**
+   * Updates a leaf at a given index in the tree
+   * @param leaf The leaf value to be updated
+   * @param index The leaf to be updated
+   */
+  updateLeaf(leaf: Buffer | LeafData, index: bigint): Promise<void>;
+  /**
+   * Returns the depth of the tree
+   */
+  getDepth(): number;
```

Comment:
Do we need this? I believe we should be able to get it from the constants, right?

Reply:
It can be returned from the constants, however it made sense for me that `getTreeInfo` to returns its own depth,  this method was added to enable it.
I can remove if its clutter